src/main/java/org/wso2/carbon/inbound/iso8583/listening/ConnectionRequestHandler.java

```diff
@@ -61,7 +63,14 @@ public void connect() throws IOException {
             int messageLength = inputStreamReader.available();
             byte[] message = new byte[messageLength];
             inputStreamReader.readFully(message, 0, messageLength);
+            ISO8583Version iso8583Version = getISO8583Version(message);
```

Comment:
In ISO8583 standard messages that starts from 0 are v1987 and starts with 1 are v1993 and so on. Currently our implementation does not check whether the given message version but assume its v1987 and proceeds. I have added this method to identify the version of the message and use the packager according to the version. With this change we are only supporting v1987 and v1993 and any other version messages will cause the unsupported exception.
`Will this run fully backward compatible manner?`
Yeah, it will support both v1987 and v1993

`Will it be possible to assume the default ISO version is 1987 and only switch to 1993 if we detect that in the message?`
Yeah we can change this getISO8583Version method to do so. But if someone uses a version that is not supported they will either get a parse error or a wrong encoding.